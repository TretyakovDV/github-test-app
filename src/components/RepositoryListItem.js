import React from 'react';
import PropTypes from 'prop-types';

const RepositoryListItem = ({ title, stars, url }) => (
  <li>
    <span>{title}</span>
    <span>{stars}</span>
    <a href={url}>Go to repository</a>
  </li>
);

RepositoryListItem.propTypes = {
  title: PropTypes.string.isRequired,
  stars: PropTypes.number.isRequired,
  url: PropTypes.string.isRequired,
};

export default RepositoryListItem;
