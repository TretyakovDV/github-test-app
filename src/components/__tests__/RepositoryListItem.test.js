import React from 'react';
import { render } from '@testing-library/react';
import RepositoryListItem from '../RepositoryListItem';

describe('RepositoryListItem', () => {
  it('matches snapshot', () => {
    const title = 'title';
    const stars = 200;
    const url = 'url';

    const { asFragment } = render(<RepositoryListItem title={title} stars={stars} url={url} />);
    expect(asFragment()).toMatchSnapshot();
  });
});
