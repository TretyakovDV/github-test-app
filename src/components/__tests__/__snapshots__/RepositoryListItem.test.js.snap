// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddRepositoryForm matches snapshot 1`] = `
<DocumentFragment>
  <li>
    <span>
      title
    </span>
    <span>
      200
    </span>
    <a
      href="url"
    >
      Go to repository
    </a>
  </li>
</DocumentFragment>
`;
